### PR TITLE
Users pagination

### PR DESCRIPTION
**Pull request by JJBai2003**
## Summary

1. Added new endpoint GET /users/pagination
2. Implemented getUsersPaginated in /users/controller.ts
3. Populated test cases for getUsersPaginated
4. Tested getUsersPaginated using Postman to double-check check function's correctness

### Testing using Jest
1. Run yarn backend
2. Run yarn test
Screen Shot 2023-03-18 at 4 41 49 PM

### Testing using Postman
1. Starting a GET request in the form of "http://localhost:8000/users/paginations?limit=<number>&after=<userid>"
2. SEND and we should see the correct number of users and the correct users displayed

### CODE REVIEW

File: backend/src/users/controllers.ts [hunk 1]
```diff
@@ -178,6 +178,33 @@ const getAllUsers = async (req: Request, res: Response) => {
     res.status(500).json({ error: (error as Error).message });
   }
 };
+
+/**
+ * Gets all Users in database with pagination
+ * @returns promise with all users or error
+ *
+ *
```
Comment (Archit404Error) [inline backend/src/users/controllers.ts:186]: Remove the extra comment lines:
```suggestion
```

File: backend/src/users/controllers.ts [hunk 2]
```diff
@@ -178,6 +178,33 @@ const getAllUsers = async (req: Request, res: Response) => {
...
+        id: query.after ? query.after as string : undefined,
+      },
+      orderBy: {
+        id: "asc",
+      },
+    }
+    );
+    res.status(200).json(users);
+  } catch (error) {
+    res.status(500).json({ error: (error as Error).message });
```
Comment (Archit404Error) [inline backend/src/users/controllers.ts:204]: You don't have to do this but ideally we should have a `utils/` folder with success and error json methods to standardize error responses

File: backend/tests/users.test.ts [hunk 3]
```diff
@@ -429,3 +429,23 @@ describe("Testing DELETE user", () => {
     expect(response.status).toBe(500);
   });
 });
+
+
+describe("Testing GET /users/pagination", () => {
+  test("Get 2 users after the second use", async () => {
+    const users = await request(app).get("/users");
+    const userid = users.body[1].id;
+    const response = await request(app).get("/users/pagination?limit=2&after=" + userid);
```
Comment (Archit404Error) [inline backend/tests/users.test.ts:438]: Small syntactic sugar suggestion:
```suggestion
    const response = await request(app).get(`/users/pagination?limit=2&after=${userid}`);
```

File: backend/tests/users.test.ts [hunk 4]
```diff
@@ -429,3 +429,23 @@ describe("Testing DELETE user", () => {
...
+    const data = response.body;
+    expect(response.status).toBe(200);
+    expect(data.length).toBe(2);
+  });
+  
+  test("Get 10 users after the second use", async () => {
+    // limit should be defaulted to 10
+    const users = await request(app).get("/users");
+    const userid = users.body[2].id;
+    const response = await request(app).get("/users/pagination?after=" + userid);
```
Comment (Archit404Error) [inline backend/tests/users.test.ts:448]: More sugar:
```suggestion
    const response = await request(app).get(`/users/pagination?after=${userid}`);
```

Alternatively, you can also just remove line 447 entirely and do ```const response = await request(app).get(`/users/pagination?after=${users.body[2].id}`);``` but I'll leave that decision up to you

**Review by Archit404Error — APPROVED** [review]
Good job on this PR! Just left a few style fixes and sugar suggestions -- don't merge until one of the TLs approve as well though!

**Comment by bananahz** [comment]
lgtm :)

**Review by bananahz — APPROVED** [review]
lgtm

**Comment by danielbotros** [comment]
LGTM!

**Pull request merged**